trieste/acquisition/function.py

```diff
@@ -519,3 +539,78 @@ def prepare_acquisition_function(
         :param models: The models over each dataset in ``datasets``.
         :return: A batch acquisition function.
         """
+
+
+class BatchReparametrizationSampler:
+    r"""
+    This sampler employs the *reparameterization trick* to approximate batches of samples from a
+    :class:`ProbabilisticModel`\ 's predictive joint distribution as
+
+    .. math:: x \mapsto \mu(x) + \epsilon L(x)
+
+    where :math:`L` is the Cholesky factor s.t. :math:`LL^T` is the covariance, and
+    :math:`\epsilon \sim \mathcal N (0, 1)` is constant for a given sampler, thus ensuring samples
+    form a continuous curve.
+    """
+
+    def __init__(self, sample_size: int, model: ProbabilisticModel):
+        """
+        :param sample_size: The number of samples for each batch of points. Must be positive.
+        :param model: The model to sample from.
+        :raise ValueError (or InvalidArgumentError): If ``sample_size`` is not positive.
+        """
+        tf.debugging.assert_positive(sample_size)
+
+        self._sample_size = sample_size
+
+        # _eps is essentially a lazy constant, declared here but initialised only on the first call
```

Comment:
I know what you mean, but arguably the `_eps` is initialised here since it has been assigned a value, even if it is a dummy value. Perhaps something like "declared and assigned a dummy value here, the actual value is assigned on the first call to sample" would be clearer?

Reply:
this comment is mainly for @vpicheny , so to some extent it depends which he finds more helpful

Reply:
how about
```
        # _eps is essentially a lazy constant. It is declared and assigned an empty tensor here, and
        # populated on the first call to sample
```